cpp/src/io/json/json_tree.cu

```diff
@@ -237,6 +240,15 @@ tree_meta_t get_tree_representation(device_span<PdaTokenT const> tokens,
     };
   };
 
+  // Look for ErrorBegin and report the point of error.
+  if (auto error_location =
+        thrust::find(rmm::exec_policy(stream), tokens.begin(), tokens.end(), token_t::ErrorBegin);
```

Comment:
In addition, `thrust::find` is well known for performance regression, which has been mentioned in may places in cudf discussions. Please use `thrust::count_if() > 0`.

Reply:
Note that if we want to print the content of the found location then `thrust::find` is unavoidable :(

Reply:
If the issues become noticeable, you could simulate it with a min reduction over a counting transform iterator
```cpp
[tokens] __device__ (size_t i) { return tokens[i] == token_t::ErrorBegin ? i : tokens.size(); }
```

Reply:
I vaguely recall having heard this before. Do you know the reason why this is the case? Wondering how large the difference is between the two. 

@karthikeyann, I'm wondering if we could fuse this with some other reduction over the tokens and save doing this extra pass.

Reply:
To optimize for the common case (non-error case), we could use `thrust::count`, then for error-case, `thrust::find` can be used later to find the exact location of the error. The exact location is needed, otherwise the user would be clueless about the location of the error in input file/string.